embedx/config.schema.json

```diff
@@ -1963,9 +1963,29 @@
             }
           },
           "additionalProperties": false
+        },
+        "channels": {
+          "type": "array",
+          "items": {
+            "title": "Courier channel configuration",
+            "type": "object",
+            "properties": {
+              "id": {
+                "type": "string",
+                "title": "Channel id",
+                "description": "The channel id. Corresponds to the .via property of the identity schema for recovery, verification, etc. Currently only phone is supported.",
+                "maxLength": 255,
+                "enum": ["phone"]
+              },
+              "request_config": {
+                "$ref": "#/definitions/httpRequestConfig"
+              }
+            },
+            "required": ["id", "request_config"],
+            "additionalProperties": false
+          }
         }
       },
-      "required": ["smtp"],
```

Comment:
From now on it's no longer required to set up an SMTP server in the config. What will happen if we don't define an SMTP server here?

Reply:
People won't complain about having to set an SMTP URL, even if you only use the HTTP based sender ;)

I can't think of any implications. If you don't set the connection URL, instantiating the courier will return an error here: https://github.com/ory/kratos/blob/c0e528f3408e001cd29e09406094723e09a48147/courier/smtp.go#L30

And if you don't set any other delivery config, Kratos won't start, because we fall back to the SMTP "channel" in that case. 

So essentially, instead of catching this "misconfiguration" in the config validation is caught in the bootstrapping logic of Kratos, which doesn't lead to any change in the behavior of Kratos as it won't start either way.

Reply:
The bootstraping logic is different in Ory Network, will that have negative consequences there?

Reply:
Only for the backoffice courier. Which I have adapted already.

Reply:
Awesome!